app/src/main/java/space/taran/arkbrowser/ui/adapter/PreviewsPager.kt

```diff
@@ -22,8 +25,12 @@ class PreviewsPager(val presenter: PreviewsList)
     override fun onBindViewHolder(holder: PreviewItemViewHolder, position: Int) {
         holder.pos = position
         presenter.bindView(holder)
-        holder.setOnClickHandler {
-            presenter.itemClicked(position)
+        val gestureDetector = getGestureDetector(holder)
+        holder.itemView.layout_root.setOnTouchListener { view, motionEvent ->
```

Comment:
Just curious, why `GestureDetector` is better?

Reply:
OnClickListener is triggered at the end of the swipe, in this case it is impossible to use the immersive mod.